File: oc-bridge-service/src/main/java/service/OcBridge.java
package service;

import io.vertx.core.AbstractVerticle;
import io.vertx.ext.web.Router;
import io.vertx.ext.web.handler.BodyHandler;

public class OcBridge extends AbstractVerticle {
    private static final int PORT = 10000;
    private static final String HOST = "localhost";

    @Override
    public void start() {
        vertx
                .createHttpServer()
                .requestHandler(createRouter())
                .listen(PORT, HOST);
        System.out.println("Service ready.");
    }

    private Router createRouter() {
        RequestManager.initializeRequestManager(vertx);
        Router router = Router.router(vertx);
        router.route().handler(BodyHandler.create());
        router.post("/v1/events").handler(RequestManager::handleCreateEvent);
        router.get("/v1/events/:eventId").handler(RequestManager::handleGetEventById);
        router.patch("/v1/events/:eventId").handler(RequestManager::handleUpdateEvent);
        return router;
    }
}

File: oc-bridge-service/src/main/kotlin/service/OcBridge.kt
package service

import io.vertx.core.AbstractVerticle
import io.vertx.ext.web.Router
import io.vertx.ext.web.handler.BodyHandler

class OcBridge : AbstractVerticle() {

    override fun start() {
        vertx
                .createHttpServer()
                .requestHandler(createRouter())
                .listen(PORT, HOST)
        println("Service ready on port $PORT and host $HOST")
    }

    private fun createRouter(): Router {
        RequestManager.initializeRequestManager(vertx)
        return Router.router(vertx).apply {
            route().handler(BodyHandler.create())
            post(EVENTS_PATH).handler { RequestManager.createEvent(it) }
            get(SINGLE_EVENT_PATH).handler { RequestManager.retrieveEventById(it) }
            patch(SINGLE_EVENT_PATH).handler { RequestManager.updateEvent(it) }
        }
    }

    companion object {
        private const val PORT = 10000
        private const val HOST = "localhost"
        private const val EVENTS_PATH = "/v1/events"
        private const val SINGLE_EVENT_PATH = "$EVENTS_PATH/:eventId"
    }
}
